File: core/java/src/main/java/org/aiddl/core/function/misc/QuoteFunction.java
package org.aiddl.core.function.misc;

import org.aiddl.core.interfaces.LazyFunction;
import org.aiddl.core.representation.Term;

public class QuoteFunction implements LazyFunction {

	@Override
	public Term apply(Term x) {
		return x;
	}

}

File: core/scala/src/main/scala/org/aiddl/core/scala/function/misc/QuoteFunction.scala
package org.aiddl.core.scala.function.misc

import org.aiddl.core.scala.function.{Function, LazyFunction}
import org.aiddl.core.scala.representation.Term

protected[function] class QuoteFunction extends  Function with LazyFunction {
  def apply(x: Term): Term = x
}
